pipes/WDL/tasks/tasks_16S_amplicon.wdl

```diff
@@ -0,0 +1,313 @@
+        #Make a manifest.txt that contains [1.sample-id 2.R1_fastq 3.R2.fastq]
+        #> =overwrite or writes new file 
+        echo -e "sample-id\tforward-absolute-filepath\treverse-absolute-filepath" > manifest.tsv
+        #>>= appends 
+        #\t= tabs next value 
+        echo -e "$NEWSAMPLENAME\t$(pwd)/R1.fastq.gz\t$(pwd)/R2.fastq.gz" >> manifest.tsv
+        
+        #fastq -> bam (provided by qiime tools import fxn)
+        qiime tools import \
+            --type 'SampleData[PairedEndSequencesWithQuality]' \
+            --input-path manifest.tsv \
+            --input-format PairedEndFastqManifestPhred33V2 \
+            --output-path "~{sample_name}.qza"
+    >>>
+
+    output {
+        File   reads_qza               = "~{sample_name}.qza"
+        String samplename_master_sheet = read_string("NEWSAMPLENAME.txt")
+    }
+
+    runtime {
+        docker: "~{docker}"
+        memory: "1GB"
+        cpu: 1 
+    }
+}
+
+#--------------------------------------------
+#Part 1 | Step 2:cutadapt: Trim sequences 
+#trimreads_trim
+#trim = default 
+task trim_reads {
+
+    meta {
+        description:"Removing adapter sequences, primers, and other unwanted sequence from sequence data."
+    }
+
+    input {
+        File    reads_qza
+        
+        String  qza_basename = basename(reads_qza, '.qza')
+        #Boolean not_default = false
+        String  forward_adapter         = "CTGCTGCCTCCCGTAGGAGT"
+        String  reverse_adapter         = "AGAGTTTGATCCTGGCTCAG"
+        Int?    min_length              = 1
+        Boolean? keep_untrimmed_reads   = false
+
+        String  docker          = "quay.io/qiime2/core:2022.8" 
+    }
+
+    command <<<
+        set -ex -o pipefail
+        # obtain the name of the qiime conda environment in the container; flexible about version
+        CONDA_ENV_NAME=$(conda info --envs -q | awk -F" " '/qiime.*/{ print $1 }')
+        # activate the qiime conda environment
+        # seemingly necessary because of:
+        #   https://github.com/chanzuckerberg/miniwdl/issues/603
+        conda activate ${CONDA_ENV_NAME}
+
+        qiime cutadapt trim-paired \
+        --i-demultiplexed-sequences "~{reads_qza}" \
+        --p-front-f "~{forward_adapter}" \
+        --p-front-r "~{reverse_adapter}" \
+        ~{"--p-minimum-length " + min_length} \
+        ~{true='--p-no-discard-untrimmed' false='--p-discard-untrimmed' keep_untrimmed_reads} \
+        --o-trimmed-sequences "~{qza_basename}_trimmed.qza"
+
+        #trim_visual 
+        qiime demux summarize \
+        --i-data "~{qza_basename}_trimmed.qza" \
+        --o-visualization "~{qza_basename}_trim_summary.qzv"
+    >>>
+
+    output {
+        #trimmed_sequences = paired ends for vsearch
+        File trimmed_reads_qza     = "~{qza_basename}_trimmed.qza"
+        File trimmed_visualization = "~{qza_basename}_trim_summary.qzv" 
+    }
+
+    runtime {
+        docker: "~{docker}"
+        memory: "10 GB"
+        cpu: 1
```

Comment:
We'll look more closely at ram/cpu sometime later after this PR merges -- no need to optimize that for this PR.

Reply:
Ok sounds good